compiler/rustc_const_eval/src/interpret/intern.rs

```diff
@@ -393,15 +433,29 @@ pub fn intern_const_alloc_recursive<
     debug!(?todo);
     debug!("dead_alloc_map: {:#?}", ecx.memory.dead_alloc_map);
     while let Some(alloc_id) = todo.pop() {
-        if let Some((_, mut alloc)) = ecx.memory.alloc_map.remove(&alloc_id) {
+        if let Some((kind, mut alloc)) = ecx.memory.alloc_map.remove(&alloc_id) {
+            for &(_, alloc_id) in alloc.provenance().ptrs().iter() {
+                if leftover_allocations.insert(alloc_id) {
+                    todo.push(alloc_id);
+                }
+            }
```

Comment:
Why is this being reordered?

Reply:
Because of the early returns in the static case